RockWeb/Blocks/Connection/ConnectionOpportunitySignup.ascx.cs

```diff
@@ -351,10 +351,10 @@ public void ShowDetail( int opportunityId )
                     return;
                 }
 
-                if ( !opportunity.IsActive )
+                if ( !opportunity.IsActive || !opportunity.ConnectionType.IsActive )
                 {
                     pnlSignup.Visible = false;
-                    ShowError( "Inactive Opportunity Type", "The opportunity is not currently active." );
+                    ShowError( "Inactive Opportunity Type", "The opportunity or connection type is not currently active." );
```

Comment:
Since this message is intended for a normal, non-administrator type person, I think leaving the message as simple as possible is better.  (And this is not your fault, but I'm not even sure it should be using "Inactive Opportunity Type" as the title to that error message.)

There's a new section called "_Coding for the Target Persona_" in our [DRAFT 303 Developer Guide](https://community.rockrms.com/Developer/draft303) that briefly covers this point.

Reply:
So maybe this should say something along the lines of

```suggestion
                    ShowError( "Sorry ", "The are no current opportunities." );
```

Also in line 350 there is the same wording, should that be changed as well?

Reply:
Should read, "**There** are no current opportunities." Correct?

Reply:
Oops, yes that is correct.